design/httpproxy-header-rewrite.md

```diff
@@ -0,0 +1,79 @@
+# Header Rewrite For HTTPProxy
```

Comment:
This file is now defunct (in this PR) and the source of truth is: https://github.com/projectcontour/contour/pull/2066